File: 백준/Silver/1920. 수 찾기/수 찾기.swift
import Foundation

let n = Int(readLine()!)!
let nArray = readLine()!.split(separator: " ").map { Int($0)! }.sorted()
let m = Int(readLine()!)!
let mArray = readLine()!.split(separator: " ").map { Int($0)! }

for num in mArray {
    var start = 0
    var end = nArray.count - 1
    var isExist = false
    
    while start <= end {
        let middle = (start + end) / 2
        
        if nArray[middle] == num {
            isExist = true
            break
        } else if nArray[middle] > num {
            end = middle - 1
        } else {
            start = middle + 1
        }
    }
    
    print(isExist ? "1" : "0")
}

File: 백준/Silver/1920. 수 찾기/수 찾기.py
N = int(input())
nums = list(map(int, input().split()))
M = int(input())
compareNums = list(map(int, input().split()))

nums.sort()

for num in compareNums:
    left = 0
    right = N - 1
    isExist = False
    
    while left <= right:
        middle = (left + right) // 2
        
        if num == nums[middle]:
            print(1)
            isExist = True
            break
        elif num > nums[middle]:
            left = middle + 1
        else:
            right = middle - 1
            
    if not isExist:
        print(0)
    